2022/incoming/st-nicks-reindeers-need-water.pod

```diff
@@ -0,0 +1,152 @@
+was reminded of a module he recently added to the "Good List". It
+is called L<Util::H2O> and it a relatively new addition to the
+growing list of Perl modules on CPAN that turn C<HASH> references to
+blessed references with automatically generated accessor methods.
+
+Santa has, in fact, been sorely tempted to move this module to
+the C<Very Good Module List> since he's found it so handy with all
+of the Perl programming he does that involves C<HASH> references. He
+finds it particularly useful when wrangling results from L<DBI>'s
+C<selectall_hashref> method, the results produced from his web
+scraping activities using L<Web::Scraper>, or the C<HASH> reference
+he gets after C<decode_json>'ing a block of JSON data from the
+different web API he uses to track the online activities everyone
+on his various I<watch> lists.
+
+Just recently, Santa created this script that uses C<Util::H2O> to
+add accessors to data I<in flight> in order to make it easier to
+manage the response C<HASH> reference returned by L<HTTP::Tiny>.
+
+  use strict;
+  use warnings;
+  use JSON       qw//;
+  use HTTP::Tiny qw//;
+  use Util::H2O; # only exports 'h2o'
+  
+Santa can slip C<h2o> inline to his C<GET> request with more ease than
+he's able to slip into a chimney! Can I<you> spot the addition of C<h2o>?
+
+  # give's Santa "$response->content", "$response->status", "$response->success", etc
+  # from HTTP::Tiny's response object (pure HASH)
+  
+  my $http = HTTP::Tiny->new;
+  my $response = h2o $http->get(q{https://jsonplaceholder.typicode.com/users});
+
+Checking for success is clean and easy,
+
+  # check for unsuccessful web request
+
+  if (not $response->success) {
+  
+    print STDERR qq{Can't get list of online persons to watch!\n};
+    printf STDERR qq{Web request responded with with HTTP status: %d\n}, $response->status;
+    exit 1;
+  
+  }
+
+The JSON C<content> in the I<HTTP> call is also easily accessed so that
+it may be turned into a Perl data structure via C<JSON::decode_json>.
+
+  # decode JSON from response content
+  my $json_array_ref = JSON::decode_json($response->content); # $json is an ARRAY reference
+
+In this case and after referencing the API response specification, Santa
+can see that he gets an C<ARRAY> of C<HASH> references:
+
+  # [
+  #  {
+  #    "id": 1,
+  #    "name": "Leanne Graham",
+  #    "username": "Bret",
+  #    "email": "Sincere@april.biz",
+  #    "address": {
+  #      "street": "Kulas Light",
+  #      "suite": "Apt. 556",
+  #      "city": "Gwenborough",
+  #      "zipcode": "92998-3874",
+  #      "geo": {
+  #        "lat": "-37.3159",
+  #        "lng": "81.1496"
+  #      }
+  #    }
+  #  },
+  # ...
+  # ]
+  
+And even though C<h2o> only works on C<HASH> references, this provides
+I<yet another> opportunity for Santa to use C<h2o> on each I<item> in
+the C<ARRAY>, which makes dealing with each person's record much more
+jolly:
+  
+  print qq{lat, lng, name, Twitter\n};
```

Comment:
Twitter is the only one of these keys which I don't see in the example JSON.

Reply:
I am just removing this referring to a particular service. Will just be referring generally to _username_. Sorry about the confusion.